apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import sinon from 'sinon';
+import {expect} from '../../util/reconfiguredChai';
+import {mount} from 'enzyme';
+import JavalabEditor from '@cdo/apps/javalab/JavalabEditor';
+import {Provider} from 'react-redux';
+import {
+  getStore,
+  registerReducers,
+  stubRedux,
+  restoreRedux
+} from '@cdo/apps/redux';
+var filesApi = require('@cdo/apps/clientApi').files;
+import project from '@cdo/apps/code-studio/initApp/project';
+import javalab from '@cdo/apps/javalab/javalabRedux';
+
+describe('Java Lab Editor Test', () => {
+  let defaultProps, store;
+
+  beforeEach(() => {
+    stubRedux();
+    registerReducers({javalab});
+    store = getStore();
+    sinon.stub(filesApi, 'putFile');
+    sinon.stub(filesApi, 'renameFile');
+    defaultProps = {
+      onCommitCode: () => project.autosave()
+    };
+  });
+
+  afterEach(() => {
+    restoreRedux();
+    filesApi.putFile.restore();
+    filesApi.renameFile.restore();
+  });
+
+  const createWrapper = overrideProps => {
+    const combinedProps = {...defaultProps, ...overrideProps};
+    return mount(
+      <Provider store={store}>
+        <JavalabEditor {...combinedProps} />
+      </Provider>
+    );
+  };
+
+  describe('Rename', () => {
+    it('renames file after save clicked', () => {
```

Comment:
In general, enzyme unit tests should be pretty granular. For example, one might check that changing the state of `renameFileActive` displays an input or a button.
That could be done similar to this code snippet: https://github.com/code-dot-org/code-dot-org/blob/b598fa018617d95ea287a2496b3e07cb84fab6cc/apps/test/unit/code-studio/components/libraries/LibraryCreationDialogTest.js#L33-L36

Another might check that the click handler changes the state (those two might be combined, but the test shouldn't be much larger)
https://github.com/code-dot-org/code-dot-org/blob/b598fa018617d95ea287a2496b3e07cb84fab6cc/apps/test/unit/code-studio/components/libraries/ShareTeacherLibrariesTest.js#L61-L66